### PR TITLE
Validation memory fixes

### PR DESCRIPTION
**Pull request by benjamin-cates**

### CODE REVIEW

**Comment by benjamin-cates** [comment]
Should fix using too much memory during validation

File: classification/train.py [hunk 1]
```diff
@@ -174,32 +180,37 @@ def valid(model: Any,
...
-            pred.append(outputs.cpu().detach())
-            label.append(labels.cpu().detach())
-
-        pred = torch.cat(pred)
-        label = torch.cat(label)
-        if CONFIG.map_debug and CONFIG.model_checkpoint is not None:
-            torch.save(pred, "/".join(CONFIG.model_checkpoint.split('/')[:-1]) + '/pred.pt')
-            torch.save(label, "/".join(CONFIG.model_checkpoint.split('/')[:-1]) + '/label.pt')
+        with torch.no_grad():
+            for index, (mels, labels) in enumerate(dl):
```
Comment (Sean1572) [inline classification/train.py:201]: Double check to make sure validation dl is random shuffle

File: classification/train.py [hunk 2]
```diff
@@ -153,7 +153,12 @@ def train(model: Any,
 
         if (i != 0 and i % (CONFIG.valid_freq) == 0):
             valid_start_time = datetime.datetime.now()
-            _, _, best_valid_cmap = valid(model, valid_loader, epoch + i / len(data_loader), best_valid_cmap, CONFIG)
+            _, _, best_valid_cmap = valid(model, 
+                                          valid_loader, 
+                                          epoch + i / len(data_loader), 
+                                          best_valid_cmap, 
+                                          CONFIG.valid_dataset_ratio,
```
Comment (Sean1572) [inline classification/train.py]: Just call CONFIG.valid_dataset_ratio in the function, save us an argument space

**Review by Sean1572 — COMMENTED** [review]
Review these things

File: classification/train.py (hunk 2 above)
Comment (Sean1572) [reply classification/train.py]: Even better, check if epoch%1 = 0

Comment (Sean1572) [reply classification/train.py]: So we can do full validation at end

Comment (sprestrelski) [reply classification/train.py]: remove cmap - use map instead

File: classification/train.py (hunk 1 above)
Comment (benjamin-cates) [reply classification/train.py:201]: We good

File: classification/train.py (hunk 2 above)
Comment (benjamin-cates) [reply classification/train.py]: all righty thanks sam

**Comment by benjamin-cates** [comment]
Runs on my machine after merge commit
